apps/common/behaviours/casework-submission.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+const CaseworkModel = require('../models/i-casework');
+
+const Compose = func => superclass => class extends superclass {
+
+  prepare() {
+    if (typeof func === 'function') {
+      return Object.assign(super.prepare(), func(this.toJSON()));
+    }
+    return super.prepare();
+  }
+
+};
+
+module.exports = config => {
+
+  config = config || {};
+
+  // compose custom per-app prepare methods onto the standard model
+  const Model = Compose(config.prepare)(CaseworkModel);
+
+  return superclass => class extends superclass {
+
+    saveValues(req, res, next) {
+      const model = new Model(req.sessionModel.toJSON());
+      model.save()
+        .then(() => {
+          super.saveValues(req, res, next);
```

Comment:
I'm *guessing* we'll want to save the casework id returned and present it to the user somehow, but as with all things, requirementslol.